contrib/win32/win32compat/wmain_sshd.c

```diff
@@ -260,14 +260,40 @@ int argc_original = 0;
 wchar_t **wargv_original = NULL;
 
 int wmain(int argc, wchar_t **wargv) {
-	wchar_t* path_utf16;
+	wchar_t *path_value = NULL, *path_new_value;
+	errno_t result = 0;
+	size_t path_new_len = 0, len;
 	argc_original = argc;
 	wargv_original = wargv;
 
 	init_prog_paths();
 	/* change current directory to sshd.exe root */
 	_wchdir(__wprogdir);
 
+	_wdupenv_s(&path_value, &len, L"PATH");
+	if (!path_value || (wcsstr(path_value, __wprogdir)) == NULL) {
+		path_new_len = wcslen(__wprogdir) + wcslen(path_value) + 2;
+		if ((path_new_value = (wchar_t *) malloc(path_new_len * sizeof(wchar_t))) == NULL) {
+			errno = ENOMEM;
+			error("failed to allocation memory");
+			return -1;
+		}
+		swprintf_s(path_new_value, path_new_len, L"%s%s%s", __wprogdir, path_value ? L";" : L"",  path_value);
```

Comment:
What's the confirmed order of exe lookup (we figured that what's on msdn is wrong) ? Does this guarantee picking up exe from progdir in all scenarios?
Are we OK with adjusting PATH for user's remote sessions?